apps/docs/pages/docs/api-reference/configuration/component-config.mdx

```diff
@@ -358,6 +377,20 @@ const resolveData = async ({ props }, { changed }) => {
 
 The data object from the previous run of this function.
 
+##### `params.metadata`
+
+Global metadata provided to the `<Puck>` component. See the [`Metadata` docs](/docs/api-reference/metadata) for a full reference.
+
+```tsx copy {2-4} /changed/1 filename="Example only updating 'resolvedTitle' when 'title' changes"
+const resolveData = async ({ props }, { metadata }) => {
+  return {
+    props: { title: metadata.title || props.title },
+  };
+};
+```
+
+Note that returning metadata in the resolveData API will overwrite the prop values. For non-destructive use of metadata, refer to the [`render(puck.metadata)` docs](#puckmetadata).
+
```

Comment:
Hm I hadn't thought that we might need to expose this in `resolveData`, but I think it makes sense 🤔

Reply:
Since metadata can be anything, a value contained within could be used in resolveData as a conditional operator. For example, `if (metadata.product.type === "water") return { props: { description: "It's very wet" } };`